### PR TITLE
smt2 parser: added declare-var

### PR DESCRIPTION
**Pull request by kroening**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/solvers/smt2/smt2_parser.cpp [hunk 1]
```diff
@@ -1135,6 +1135,28 @@ void smt2_parsert::command(const std::string &c)
       return;
     }
 
+    auto &entry = id_map[id];
+    entry.type = sort();
+    entry.definition = nil_exprt();
+  }
+  else if(c == "declare-var") // this appears to be accepted by Z3 and CVC4
```
Comment (tautschnig) [inline src/solvers/smt2/smt2_parser.cpp]: Why not just do `if(c == "declare-const" || c == "declare-var")` above and then use `c` in the error message?

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4af5066d02a0872b67ccdd4f29e79b2d11fcdfa0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91501254

File: src/solvers/smt2/smt2_parser.cpp (hunk 1 above)
Comment (kroening) [reply src/solvers/smt2/smt2_parser.cpp]: sure, done

File: src/solvers/smt2/smt2_parser.cpp [hunk 2]
```diff
@@ -1117,11 +1117,13 @@ typet smt2_parsert::function_signature_declaration()
 
 void smt2_parsert::command(const std::string &c)
 {
-  if(c=="declare-const")
+  if(c=="declare-const" || c == "declare-var")
```
Comment (tautschnig) [inline src/solvers/smt2/smt2_parser.cpp]: Add spaces around `==` to make clang-format happy.

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e58af2bb5dc3d09248296f75b1451fa9bdc1464c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91583232

**Pull request merged**